### PR TITLE
Authentication + Profile Creation

### PR DESCRIPTION
**Pull request by WesLeeO**
Login and SignUp features implemented with tests.
Firebase is storing users and their profiles.

### CODE REVIEW

**Comment by Eliasgoat** [comment]
Make sure the code has at least 80% on line coverage

File: app/src/main/java/com/github/se/gatherspot/model/Category.kt [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  DEFAULT_CATEGORY_4,
+  DEFAULT_CATEGORY_5,
+  DEFAULT_CATEGORY_6,
+  DEFAULT_CATEGORY_7,
+  DEFAULT_CATEGORY_8,
+  DEFAULT_CATEGORY_9,
+  DEFAULT_CATEGORY_10,
+  DEFAULT_CATEGORY_11,
+  DEFAULT_CATEGORY_12,
+  DEFAULT_CATEGORY_13
```
Comment (Eliasgoat) [inline app/src/main/java/com/github/se/gatherspot/model/Category.kt:27]: Replace these from the list we sent (or make sure to not use it when Melvin's class get merged)

File: app/src/main/java/com/github/se/gatherspot/model/Profile.kt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+package com.github.se.gatherspot.model
+
+data class Profile(val interests: Set<Category>)
```
Comment (Eliasgoat) [inline app/src/main/java/com/github/se/gatherspot/model/Profile.kt:3]: To be removed when merging

**Review by Eliasgoat — COMMENTED** [review]
Firebase Review Done. UI tests are to be implemented in another PR. We can confirm the merge when Sam finished the UI reviews.

**Review by SamBachmann — APPROVED** [review]
Clean interface. Good job !

File: app/src/main/java/com/github/se/gatherspot/ui/SignUp.kt
Comment (SamBachmann) [inline app/src/main/java/com/github/se/gatherspot/ui/SignUp.kt:1]: Clean interface 👍🏼

**Comment by schuetzcarl** [comment]
@Mathunter2 @WesLeeO Please, could you indicate more precisely what were your individual contributions to that PR ?

**Pull request merged**